src/clj/game/cards/programs.clj

```diff
@@ -1139,19 +1139,19 @@
              :effect (effect (trash eid card {:cause :purge}))}]})
 
 (defcard "Expert Schedule Analyzer"
-  {:abilities [{:cost [:click 1]
-                :msg "make a run on HQ"
-                :makes-run true
-                :async true
-                :effect (effect (make-run eid :hq card))}]
-   :events [(successful-run-replace-breach
-              {:target-server :hq
-               :this-card-run true
-               :ability
-               {:msg (msg "reveal all of the cards cards in HQ: "
-                          (string/join ", " (map :title (:hand corp))))
-                :async true
-                :effect (effect (reveal eid (:hand corp)))}})]})
+  (let [ability (successful-run-replace-breach
+                 {:target-server :hq
+                  :ability
+                  {:msg (msg "reveal all of the cards cards in HQ: "
+                             (string/join ", " (map :title (:hand corp))))
+                   :async true
+                   :effect (effect (reveal eid (:hand corp)))}})]
+    {:abilities [{:cost [:click 1]
+                  :msg "make a run on HQ"
+                  :makes-run true
+                  :async true
+                  :effect (effect (register-events card [(assoc ability :duration :end-of-run)])
```

Comment:
You can just put the duration directly on the ability. (Same with the other card abilities in this PR.) the ability maps aren't used elsewhere so you don't have to worry about the duration leaking into another usage.